src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -0,0 +1,1470 @@
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp.Emit;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal abstract class SourcePropertySymbolBase : PropertySymbol, IAttributeTargetSymbol
+    {
+        /// <summary>
+        /// Condensed flags storing useful information about the <see cref="SourcePropertySymbolBase"/>
+        /// so that we do not have to go back to source to compute this data.
+        /// </summary>
+        [Flags]
+        private enum Flags : byte
+        {
+            IsExpressionBodied = 1 << 0,
+            IsAutoProperty = 1 << 1,
+            IsExplicitInterfaceImplementation = 1 << 2,
+        }
+
+        // TODO (tomat): consider splitting into multiple subclasses/rare data.
+
+        private readonly SourceMemberContainerTypeSymbol _containingType;
+        private readonly string _name;
+        private readonly SyntaxReference _syntaxRef;
+        protected readonly DeclarationModifiers _modifiers;
+        private readonly ImmutableArray<CustomModifier> _refCustomModifiers;
+        private readonly SourcePropertyAccessorSymbol _getMethod;
+        private readonly SourcePropertyAccessorSymbol _setMethod;
+        private readonly TypeSymbol _explicitInterfaceType;
+        private readonly ImmutableArray<PropertySymbol> _explicitInterfaceImplementations;
+        private readonly Flags _propertyFlags;
+        private readonly RefKind _refKind;
+
+        private SymbolCompletionState _state;
+        private ImmutableArray<ParameterSymbol> _lazyParameters;
+        private TypeWithAnnotations.Boxed _lazyType;
+
+        /// <summary>
+        /// Set in constructor, might be changed while decoding <see cref="IndexerNameAttribute"/>.
+        /// </summary>
+        protected readonly string _sourceName;
+
+        private string _lazyDocComment;
+        private string _lazyExpandedDocComment;
+        private OverriddenOrHiddenMembersResult _lazyOverriddenOrHiddenMembers;
+        private SynthesizedSealedPropertyAccessor _lazySynthesizedSealedAccessor;
+        private CustomAttributesBag<CSharpAttributeData> _lazyCustomAttributesBag;
+
+        // CONSIDER: if the parameters were computed lazily, ParameterCount could be overridden to fall back on the syntax (as in SourceMemberMethodSymbol).
+
+        public Location Location { get; }
+
+#nullable enable
+        /// <remarks>
+        /// The <paramref name="bodyBinder" /> is only required for explicit interface implementations or
+        /// overrides, in which case it is passed to <see cref="ComputeType" />
+        /// </remarks>
+        protected SourcePropertySymbolBase(
+           SourceMemberContainerTypeSymbol containingType,
+           Binder? bodyBinder,
```

Comment:
>bodyBinder [](start = 19, length = 10)

I understand that this is is preexisting name, however it doesn't actually accurately reflect what binder is that, or, at least, likely to confuse a reader. I believe this is actually a type body binder, it is not specific to the property or any member in the type, especially not to any member body. Consider renaming accordingly